src/blockchain/merkle-patricia.js

```diff
@@ -189,46 +209,40 @@ export function merklePatriciaProof(rootHash, proofNode, key, type) {
                 throw new Error('Invalid length of key in tree.');
             }
 
-            if (keySuffix[0] === '0') { // '0' at the beginning means left branch/leaf
-                if (levelData.left === undefined) {
-                    levelData.left = {
-                        hash: branchValueHash,
-                        key: branchKey,
-                        type: branchType,
-                        suffix: keySuffix,
-                        size: fullKey.length
-                    };
-                } else {
-                    throw new Error('Left node is duplicated in tree.');
-                }
-            } else { // '1' at the beginning means right branch/leaf
-                if (levelData.right === undefined) {
-                    levelData.right = {
-                        hash: branchValueHash,
-                        key: branchKey,
-                        type: branchType,
-                        suffix: keySuffix,
-                        size: fullKey.length
-                    };
-                } else {
-                    throw new Error('Right node is duplicated in tree.');
-                }
-            }
+            nodes.push({
+                hash: branchValueHash,
+                key: branchKey,
+                type: branchType,
+                suffix: keySuffix,
+                size: fullKey.length
+            });
+        }
+
+        var orderedNodes = orderNodes(nodes, keyPrefix.length);
```

Comment:
Yes, it's always only 2 elements at this step.
This check is already presents in the beginning of `recursive` function:
```
if (Object.keys(node).length !== 2) {
    throw new Error('Invalid number of children in the tree node.');
}
```
So I just renamed variables inside `orderNodes` function to not confuse with `left` and `right` variables names:
```
function orderNodes(nodes, depth) {
    var child1 = nodes[0];
    var child2 = nodes[1];
    ...
```